draft-rha-jose-hpke-encrypt.md

```diff
@@ -208,51 +219,47 @@ This example demonstrates the representaton of an encapsulated key as a JWK.
 
 ### Integrated Encryption
 
-In Integrated Encryption mode, HPKE is employed to directly encrypt the (compressed) plaintext, and the resulting ciphertext is included in the JWE ciphertext. 
+This mode only supports a single recipient and JWE Compact serialization.
 
-In this mode, JWE Compact serialization MUST be used. The sender MUST specify the 'epk' and 'alg' parameters in the protected header to indicate the use of HPKE. Optionally, the protected header MAY contain the 'kid' parameter used to identify the static recipient public key used by the sender. 
+HPKE is employed to directly encrypt the plaintext, and the resulting ciphertext is included in the JWE ciphertext. 
 
-The Single-Shot APIs specified in Section 6 of {{RFC9180}} for encryption and decryption cannot be used. This is because they require an 'aad' parameter, which takes the Encoded Protected Header comprising of 'ek' as input.
+The sender MUST specify the 'epk' and 'alg' parameters in the protected header to indicate the use of HPKE. 
 
-In Integrated Encryption mode, HPKE is employed to directly encrypt the plaintext, and the resulting ciphertext is included in the JWE ciphertext. 
+The 'enc' (Encryption Algorithm) parameter MUST NOT be present. 
```

Comment:
The JWE way of doing this is IMHO to use "alg":"dir" and "enc": "HPKE...". This would remove the restriction to only support compact mode (which i assume is to prevent "enc" being slapped in to an unprotected header) and comply with JWE RFC7516 section 4.1.2

Reply:
I agree

Reply:
added here: https://github.com/tireddy2/JOSE_HPKE/pull/19/files#diff-68750195941eedd2542fd99d7986f5f9b72b41fc29189ac49e31ad55bb196240R256